src/ngraph/pass/nop_elimination.cpp

```diff
@@ -35,13 +39,37 @@ using namespace ngraph;
 
 #define TI(x) std::type_index(typeid(x))
 
+static bool remove_update_name(const std::shared_ptr<Node>& node,
```

Comment:
Since you're in this file anyways, could you change the map key. to `Node::type_info_t` and change `TI` to be `x::type_info` ? The table lookup then uses `node->get_type_info()` to look up the handler instead of `TI(node)`.